updater/lib/dependabot/updater/operations/refresh_version_update_pull_request.rb

```diff
@@ -64,9 +64,23 @@ def perform
           @notices = dependency_snapshot.notices
           # More notices can be added during the update process
 
-          check_and_update_pull_request(dependencies)
-        rescue StandardError => e
-          error_handler.handle_dependency_error(error: e, dependency: dependency)
+          # Set has_update to nil to indicate that we don't know if there is an update yet
+          has_update = T.let(nil, T.nilable(T::Boolean))
+
+          begin
+            # has_update indicates the result of checking for updates:
+            # - true if an update is available
+            # - false if no update is needed
+            # - nil if an error occurred
+            has_update = check_and_update_pull_request(dependencies)
+          rescue StandardError => e
+            error_handler.handle_dependency_error(error: e, dependency: dependency)
+          ensure
+            # Record ecosystem metrics if:
+            # - an update is available (has_update = true), or
+            # - an error occurred (has_update = nil)
+            service.record_ecosystem_meta(dependency_snapshot.ecosystem) if has_update != false
+          end
```

Comment:
**Review Tip:** Send metrics for `refresh version update` after update process (success, or fail). Just being sure that metrics is getting called when there is update. We don't prefer to send metrics when there is not update necessary